packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -820,30 +798,18 @@ describe('NetworkController', () => {
                 buildFakeClient(fakeProviders[0]),
                 buildFakeClient(fakeProviders[1]),
               ];
-              mockCreateNetworkClient()
-                .calledWith({
-                  configuration: {
-                    chainId: '0x1337',
-                    rpcUrl: 'https://test.network',
-                    ticker: 'TEST',
-                    type: NetworkClientType.Custom,
-                  },
-                  fetch,
-                  btoa,
-                })
-                .mockReturnValue(fakeNetworkClients[0])
-                .calledWith({
-                  configuration: {
-                    chainId: infuraChainId,
-                    infuraProjectId,
-                    network: infuraNetworkType,
-                    ticker: infuraNativeTokenName,
-                    type: NetworkClientType.Infura,
-                  },
-                  fetch,
-                  btoa,
-                })
-                .mockReturnValue(fakeNetworkClients[1]);
+              createNetworkClientMock.mockImplementation(
```

Comment:
There were tons of tests that were doing an exact match when mocking the call to `createNetworkClient`. We don't need to do this; in most cases we can get by with a partial match. I took the opportunity to clean this up as I was updating these tests. This also allows us to not worry about whether the RPC endpoints that we use to set up the test are configured with failover URLs or not.